custom/icds_reports/reports/service_delivery_report.py

```diff
@@ -0,0 +1,168 @@
+from corehq.apps.locations.models import SQLLocation
+from custom.icds_reports.utils import india_now, DATA_NOT_ENTERED
+from custom.icds_reports.models.views import ServiceDeliveryReportView
+import copy
+
+
+class ServiceDeliveryReport(object):
+    def __init__(self, config, location, beta=False):
+        self.location = location
+        self.beta = beta
+        self.config = config
+        if self.config['beneficiary_category'] == 'pw_lw_children':
+            self.title = 'SDR - PW,LM & Children (0-3 years)'
+        else:
+            self.title = 'SDR - Children (3-6 years)'
+
+    @property
+    def location_columns(self):
+        location_names = ['state_name', 'district_name', 'block_name', 'supervisor_name', "awc_name"]
+        return location_names[:self.config['aggregation_level']]
+
+    @property
+    def headers_and_calculation(self):
+
+        headers = [(loc_column.split('_')[0].title(), loc_column) for loc_column in
```

Comment:
nit: this might be easier to read if you just had a separate array with the names as they would be, rather than transforming the location columns. especially since the supervisor one should probably be called `sector name` (pending program side approval) as that is what we use most places for that location level